benchmark/storage/recovery_benchmark.cpp

```diff
@@ -0,0 +1,137 @@
+
+      uint64_t elapsed_ms;
+      {
+        common::ScopedTimer<std::chrono::milliseconds> timer(&elapsed_ms);
+        recovery_manager.StartRecovery();
+        recovery_manager.FinishRecovery();
+        recovered_txns += recovery_manager.GetRecoveredTxnCount();
+      }
+
+      state->SetIterationTime(static_cast<double>(elapsed_ms) / 1000.0);
+
+      // Delete test txns
+      log_manager.Start();
+      delete tested;
+      log_manager.PersistAndStop();
+
+      // Call GC 3 times to fully clean up catalog
+      recovered_catalog.TearDown();
+      auto *gc = new storage::GarbageCollector(&recovery_txn_manager, nullptr);
+      gc->PerformGarbageCollection();
+      gc->PerformGarbageCollection();
+      gc->PerformGarbageCollection();
+      delete gc;
+    }
+    state->SetItemsProcessed(recovered_txns);
+  }
+};
+
+/**
+ * Run an OLTP-like workload(5 statements per txn, 60% updates, 30% select, 10% deletes).
+ */
+// NOLINTNEXTLINE
+BENCHMARK_DEFINE_F(RecoveryBenchmark, OLTPWorkload)(benchmark::State &state) {
+  LargeSqlTableTestConfiguration config = LargeSqlTableTestConfiguration::Builder()
+                                              .SetNumDatabases(1)
+                                              .SetNumTables(1)
+                                              .SetMaxColumns(5)
+                                              .SetInitialTableSize(initial_table_size_)
+                                              .SetTxnLength(5)
+                                              .SetUpdateSelectDeleteRatio({0.6, 0.3, 0.1})
+                                              .SetVarlenAllowed(true)
+                                              .build();
+
+  RunBenchmark(&state, config);
+}
+
+/**
+ * Run transactions with a large number of updates to trigger high aborts. We use a large number of max columns to have
+ * large changes and increase the chances that txns will flush logs before aborted.
+ */
+// NOLINTNEXTLINE
+BENCHMARK_DEFINE_F(RecoveryBenchmark, HighAbortRate)(benchmark::State &state) {
+  LargeSqlTableTestConfiguration config = LargeSqlTableTestConfiguration::Builder()
+                                              .SetNumDatabases(1)
+                                              .SetNumTables(1)
+                                              .SetMaxColumns(100)
+                                              .SetInitialTableSize(initial_table_size_)
+                                              .SetTxnLength(40)
+                                              .SetUpdateSelectDeleteRatio({0.8, 0.2, 0.0})
+                                              .SetVarlenAllowed(true)
+                                              .build();
+
+  RunBenchmark(&state, config);
+}
+
+BENCHMARK_REGISTER_F(RecoveryBenchmark, OLTPWorkload)->Unit(benchmark::kMillisecond)->UseManualTime()->MinTime(3);
+
+BENCHMARK_REGISTER_F(RecoveryBenchmark, HighAbortRate)->Unit(benchmark::kMillisecond)->UseManualTime()->MinTime(10);
+
```

Comment:
Done.

Reply:
Looking at it, I think the added benchmark ends up being another high-abort one (100% updates).  Looking at the `LargeSqlTableTestObject`, it appears we do not have a mechanism for simulating concurrent inserts at this level.

We probably need to add this functionality.  I'll try to look over the existing code and see how hard it would be to add it in.

Reply:
I added inserts, so this benchmark should actually stress the way we want it to.